### PR TITLE
[Localization] localize modifier.ts apply event message & translate in Korean

### PR DESCRIPTION
**Pull request by EnochG1**
## What are the changes?
Add the option to localize message text when specific modifier applied.
 - survive damage : FOCUS_BAND (10%)
 - hit heal : SHELL_BELL
 - turn heal : LEFTOVERS
 - instant revive pokemon : REVIVER_SEED
 - money interest : COIN_CASE (each 10 turn)
 - turn held item transfer : MINI_BLACK_HOLE
 - contact held item transfer : GRIP_CLAW (10%)
 - enemy turn heal : ENEMY_HEAL

## Why am I doing these changes?
All text should be localizable.
In particular, these messages are harder for non-English users to see when messages go by quickly.

## What did change?
 - Change the code for localize > src/modifier/modifier.ts
 - Added locale files, config, localization keys in each language folders.
 - Translated them to Korean.
 - Modified hit heal, turn heal English message to match original games.
image

#### 1. En

<details>
<summary>Detail (Click to expand)</summary>

| Case | Result |
| :-----: | :-----: |
| FOCUS_BAND | image |
| SHELL_BELL | image |
| LEFTOVERS | image |
| REVIVER_SEED | image |
| COIN_CASE | image |
| MINI_BLACK_HOLE | image |
| ENEMY_HEAL | image |

</details>

#### 2. Ko

<details>
<summary>Detail (Click to expand)</summary>

| Case | Result |
| :-----: | :-----: |
| FOCUS_BAND | image |
| SHELL_BELL | image |
| LEFTOVERS | image |
| REVIVER_SEED | image |
| COIN_CASE | image |
| MINI_BLACK_HOLE | image |
| ENEMY_HEAL | image |

</details>

#### 3. De

<details>
<summary>Detail (Click to expand)</summary>

| Case | Result |
| :-----: | :-----: |
| FOCUS_BAND | image |
| SHELL_BELL | image |
| LEFTOVERS | image |
| REVIVER_SEED | image |
| COIN_CASE | image |
| MINI_BLACK_HOLE | image |
| ENEMY_HEAL | image |

</details>

#### 4. Fr

<details>
<summary>Detail (Click to expand)</summary>

| Case | Result |
| :-----: | :-----: |
| FOCUS_BAND | image |
| SHELL_BELL | image |
| LEFTOVERS | image |
| REVIVER_SEED | image |
| COIN_CASE | image |
| MINI_BLACK_HOLE | image |
| ENEMY_HEAL | image |

</details>

#### 5. Pt_BR

<details>
<summary>Detail (Click to expand)</summary>

| Case | Result |
| :-----: | :-----: |
| FOCUS_BAND | image |
| SHELL_BELL | image |
| LEFTOVERS | image |
| REVIVER_SEED | image |
| COIN_CASE | image |
| MINI_BLACK_HOLE | image |
| ENEMY_HEAL | image |

</details>

#### 6. Zh_CN

<details>
<summary>Detail (Click to expand)</summary>

| Case | Result |
| :-----: | :-----: |
| FOCUS_BAND | image |
| SHELL_BELL | image |
| LEFTOVERS | image |
| REVIVER_SEED | image |
| COIN_CASE | image |
| MINI_BLACK_HOLE | image |
| ENEMY_HEAL | image |

</details>

#### 7. Zh_TW

<details>
<summary>Detail (Click to expand)</summary>

| Case | Result |
| :-----: | :-----: |
| FOCUS_BAND | image |
| SHELL_BELL | image |
| LEFTOVERS | image |
| REVIVER_SEED | image |
| COIN_CASE | image |
| MINI_BLACK_HOLE | image |
| ENEMY_HEAL | image |

</details>

## How to test the changes?
Manually.
1. Checkout this PR and change language
2. set overrides.ts
```typescript
export const STARTING_HELD_ITEMS_OVERRIDE: Array<ModifierOverride> = [ {name: "SHELL_BELL"}, {name: "COIN_CASE"}, {name: "GRIP_CLAW"}, {name: "MINI_BLACK_HOLE"} ];
export const OPP_HELD_ITEMS_OVERRIDE: Array<ModifierOverride> = [ {name: "FOCUS_BAND"}, {name: "LEFTOVERS"}, {name: "REVIVER_SEED"}, {name: "ENEMY_HEAL"} ];
```
then start new game and watch modifier apply event.

## Checklist
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

File: src/locales/pt_BR/modifier.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+export const modifier: SimpleTranslationEntries = {
+  "surviveDamageApply": "{{pokemonNameWithAffix}} hung on\nusing its {{typeName}}!",
+  "turnHealApply": "{{pokemonNameWithAffix}} restored a little HP using\nits {{typeName}}!",
+  "hitHealApply": "{{pokemonNameWithAffix}} restored a little HP using\nits {{typeName}}!",
+  "pokemonInstantReviveApply": "{{pokemonNameWithAffix}} was revived\nby its {{typeName}}!",
+  "moneyInterestApply": "You received interest of ₽{{moneyAmount}}\nfrom the {{typeName}}!",
+  "turnHeldItemTransferApply": "{{pokemonNameWithAffix}}'s {{itemName}} was absorbed\nby {{pokemonName}}'s {{typeName}}!",
+  "contactHeldItemTransferApply": "{{pokemonNameWithAffix}}'s {{itemName}} was snatched\nby {{pokemonName}}'s {{typeName}}!",
+  "enemyTurnHealApply": "{{pokemonNameWithAffix}}\nrestored some HP!",
```
Comment (josericardo-fo) [inline src/locales/pt_BR/modifier.ts:11]:
```suggestion
  "surviveDamageApply": "{{pokemonNameWithAffix}} aguentou o tranco\nusando {{typeName}}!",
  "turnHealApply": "{{pokemonNameWithAffix}} restaurou um pouco de PS usando\nseu {{typeName}}!",
  "hitHealApply": "{{pokemonNameWithAffix}} restaurou um pouco de PS usando\nseu {{typeName}}!",
  "pokemonInstantReviveApply": "{{pokemonNameWithAffix}} foi revivido\npor {{typeName}}!",
  "moneyInterestApply": "Você recebeu um juros de ₽{{moneyAmount}}\npor causa de {{typeName}}!",
  "turnHeldItemTransferApply": "{{itemName}} de {{pokemonNameWithAffix}} foi absorvido(a)\npelo {{typeName}} de {{pokemonName}}!",
  "contactHeldItemTransferApply": "{{itemName}} de {{pokemonNameWithAffix}} foi pego(a)\npelo {{typeName}} de {{pokemonName}}!",
  "enemyTurnHealApply": "{{pokemonNameWithAffix}}\nrestaurou um pouco de PS!",
```

File: src/locales/zh_CN/modifier.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+export const modifier: SimpleTranslationEntries = {
+  "surviveDamageApply": "{{pokemonNameWithAffix}} hung on\nusing its {{typeName}}!",
+  "turnHealApply": "{{pokemonNameWithAffix}} restored a little HP using\nits {{typeName}}!",
+  "hitHealApply": "{{pokemonNameWithAffix}} restored a little HP using\nits {{typeName}}!",
+  "pokemonInstantReviveApply": "{{pokemonNameWithAffix}} was revived\nby its {{typeName}}!",
+  "moneyInterestApply": "You received interest of ₽{{moneyAmount}}\nfrom the {{typeName}}!",
+  "turnHeldItemTransferApply": "{{pokemonNameWithAffix}}'s {{itemName}} was absorbed\nby {{pokemonName}}'s {{typeName}}!",
+  "contactHeldItemTransferApply": "{{pokemonNameWithAffix}}'s {{itemName}} was snatched\nby {{pokemonName}}'s {{typeName}}!",
+  "enemyTurnHealApply": "{{pokemonNameWithAffix}}\nrestored some HP!",
```
Comment (40chyan) [inline src/locales/zh_CN/modifier.ts:11]:
```suggestion
  "surviveDamageApply": "{{pokemonNameWithAffix}}用{{typeName}}\n撑住了！",
  "turnHealApply": "{{pokemonNameWithAffix}}用{{typeName}}\n回复了体力！",
  "hitHealApply": "{{pokemonNameWithAffix}}用{{typeName}}\n回复了体力！",
  "pokemonInstantReviveApply": "{{pokemonNameWithAffix}}用{{typeName}}\n恢复了活力！",
  "moneyInterestApply": "用{{typeName}}\n获得了 ₽{{moneyAmount}} 利息！",
  "turnHeldItemTransferApply": "{{pokemonNameWithAffix}}的{{itemName}}被\n{{pokemonName}}的{{typeName}}吸收了！",
  "contactHeldItemTransferApply": "{{pokemonNameWithAffix}}的{{itemName}}被\n{{pokemonName}}的{{typeName}}夺取了！",
  "enemyTurnHealApply": "{{pokemonNameWithAffix}}\n回复了一些体力！",
```
zh_cn translation

File: src/locales/zh_TW/modifier.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+export const modifier: SimpleTranslationEntries = {
+  "surviveDamageApply": "{{pokemonNameWithAffix}} hung on\nusing its {{typeName}}!",
+  "turnHealApply": "{{pokemonNameWithAffix}} restored a little HP using\nits {{typeName}}!",
+  "hitHealApply": "{{pokemonNameWithAffix}} restored a little HP using\nits {{typeName}}!",
+  "pokemonInstantReviveApply": "{{pokemonNameWithAffix}} was revived\nby its {{typeName}}!",
+  "moneyInterestApply": "You received interest of ₽{{moneyAmount}}\nfrom the {{typeName}}!",
+  "turnHeldItemTransferApply": "{{pokemonNameWithAffix}}'s {{itemName}} was absorbed\nby {{pokemonName}}'s {{typeName}}!",
+  "contactHeldItemTransferApply": "{{pokemonNameWithAffix}}'s {{itemName}} was snatched\nby {{pokemonName}}'s {{typeName}}!",
+  "enemyTurnHealApply": "{{pokemonNameWithAffix}}\nrestored some HP!",
```
Comment (40chyan) [inline src/locales/zh_TW/modifier.ts:11]:
```suggestion
  "surviveDamageApply": "{{pokemonNameWithAffix}}用{{typeName}}\n撐住了！",
  "turnHealApply": "{{pokemonNameWithAffix}}用{{typeName}}\n回復了體力！",
  "hitHealApply": "{{pokemonNameWithAffix}}用{{typeName}}\n回復了體力！",
  "pokemonInstantReviveApply": "{{pokemonNameWithAffix}}用{{typeName}}\n回復了活力！",
  "moneyInterestApply": "用{{typeName}}\n獲得了 ₽{{moneyAmount}} 利息！",
  "turnHeldItemTransferApply": "{{pokemonNameWithAffix}}的{{itemName}}被\n{{pokemonName}}的{{typeName}}吸收了！",
  "contactHeldItemTransferApply": "{{pokemonNameWithAffix}}的{{itemName}}被\n{{pokemonName}}的{{typeName}}奪取了！",
  "enemyTurnHealApply": "{{pokemonNameWithAffix}}\n回復了一些體力！",
```
zh_tw translation

**Review by returntoice — APPROVED** [review]

File: src/locales/de/modifier.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+export const modifier: SimpleTranslationEntries = {
+  "surviveDamageApply": "{{pokemonNameWithAffix}} hung on\nusing its {{typeName}}!",
+  "turnHealApply": "{{pokemonNameWithAffix}} restored a little HP using\nits {{typeName}}!",
+  "hitHealApply": "{{pokemonNameWithAffix}} restored a little HP using\nits {{typeName}}!",
+  "pokemonInstantReviveApply": "{{pokemonNameWithAffix}} was revived\nby its {{typeName}}!",
+  "moneyInterestApply": "You received interest of ₽{{moneyAmount}}\nfrom the {{typeName}}!",
+  "turnHeldItemTransferApply": "{{pokemonNameWithAffix}}'s {{itemName}} was absorbed\nby {{pokemonName}}'s {{typeName}}!",
+  "contactHeldItemTransferApply": "{{pokemonNameWithAffix}}'s {{itemName}} was snatched\nby {{pokemonName}}'s {{typeName}}!",
+  "enemyTurnHealApply": "{{pokemonNameWithAffix}}\nrestored some HP!",
+} as const;
```
Comment (CodeTappert) [inline src/locales/de/modifier.ts:12]:
```suggestion
export const modifier: SimpleTranslationEntries = {
  "surviveDamageApply": "{{pokemonNameWithAffix}} hält mithilfe des Items {{typeName}} durch!",
  "turnHealApply": "{{typeName}} von {{pokemonNameWithAffix}} füllt einige KP auf!",
  "hitHealApply": "{{typeName}} von {{pokemonNameWithAffix}} füllt einige KP auf!",
  "pokemonInstantReviveApply": "{{pokemonNameWithAffix}} wurde durch {{typeName}} wiederbelebt!",
  "moneyInterestApply": "Du erhählst {{moneyAmount}} ₽ durch das Item {{typeName}}!",
  "turnHeldItemTransferApply": "{{itemName}} von {{pokemonNameWithAffix}} wurde durch {{typeName}} von {{pokemonName}} absorbiert!",
  "contactHeldItemTransferApply": "{{itemName}} von {{pokemonNameWithAffix}} wurde durch {{typeName}} von  {{pokemonName}} geklaut!",
  "enemyTurnHealApply": "{{pokemonNameWithAffix}} stellt einige KP wieder her!",
} as const;
```

File: src/locales/fr/modifier.ts [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+
+export const modifier: SimpleTranslationEntries = {
+  "surviveDamageApply": "{{pokemonNameWithAffix}} hung on\nusing its {{typeName}}!",
+  "turnHealApply": "{{pokemonNameWithAffix}} restored a little HP using\nits {{typeName}}!",
+  "hitHealApply": "{{pokemonNameWithAffix}} restored a little HP using\nits {{typeName}}!",
+  "pokemonInstantReviveApply": "{{pokemonNameWithAffix}} was revived\nby its {{typeName}}!",
+  "moneyInterestApply": "You received interest of ₽{{moneyAmount}}\nfrom the {{typeName}}!",
+  "turnHeldItemTransferApply": "{{pokemonNameWithAffix}}'s {{itemName}} was absorbed\nby {{pokemonName}}'s {{typeName}}!",
+  "contactHeldItemTransferApply": "{{pokemonNameWithAffix}}'s {{itemName}} was snatched\nby {{pokemonName}}'s {{typeName}}!",
+  "enemyTurnHealApply": "{{pokemonNameWithAffix}}\nrestored some HP!",
```
Comment (Adri1) [inline src/locales/fr/modifier.ts:11]:
```suggestion
  "surviveDamageApply": "{{pokemonNameWithAffix}} tient bon grâce à {{typeName}} !",
  "turnHealApply": "Les PV du {{pokemonNameWithAffix}}\nsont un peu restaurés par {{typeName}} !",
  "hitHealApply": "Les PV du {{pokemonNameWithAffix}}\nsont un peu restaurés par {{typeName}} !",
  "pokemonInstantReviveApply": "{{pokemonNameWithAffix}} a repris connaissance\navec {{typeName}} et est prêt à se battre de nouveau !",
  "moneyInterestApply": "Vous recevez {{moneyAmount}} ₽\nd’intérets de la {{typeName}} !",
  "turnHeldItemTransferApply": "{{itemName}} de {{pokemonNameWithAffix}} est absorbé\npar le {{typeName}} de {{pokemonName}} !",
  "contactHeldItemTransferApply": "{{itemName}} de {{pokemonNameWithAffix}} est volé\npar {{typeName}} de {{pokemonName}} !",
  "enemyTurnHealApply": "{{pokemonNameWithAffix}}\nrestaure un peu ses PV !",
```

**Review by Adri1 — COMMENTED** [review]
Added French (complicated because of gendered items)

**Comment by EnochG1** [comment]
> Please show german screenshots once you added my suggestion

@CodeTappert You can check it in main article.

**Comment by CodeTappert** [comment]
Hmm some of them actually sound weird. Buts its offical game text mostly. So well. xD

**Comment by EnochG1** [comment]
@Adri1 @CodeTappert @40chyan @josericardo-fo
Could you please approve PR if screenshots of main article are right?

**Review by CodeTappert — APPROVED** [review]

File: src/locales/pt_BR/modifier.ts [hunk 6]
```diff
@@ -0,0 +1,12 @@
+import { SimpleTranslationEntries } from "#app/interfaces/locales";
+
+export const modifier: SimpleTranslationEntries = {
+  "surviveDamageApply": "{{pokemonNameWithAffix}} aguentou o tranco\nusando sua {{typeName}}!",
+  "turnHealApply": "{{pokemonNameWithAffix}} restaurou um pouco de PS usando\nseu {{typeName}}!",
+  "hitHealApply": "{{pokemonNameWithAffix}} restaurou um pouco de PS usando\nsua {{typeName}}!",
+  "pokemonInstantReviveApply": "{{pokemonNameWithAffix}} foi revivido\npor seu {{typeName}}!",
```
Comment (josericardo-fo) [inline src/locales/pt_BR/modifier.ts:7]:
```suggestion
  "pokemonInstantReviveApply": "{{pokemonNameWithAffix}} foi revivido\npor sua {{typeName}}!",
```

File: src/locales/pt_BR/modifier.ts [hunk 7]
```diff
@@ -0,0 +1,12 @@
+import { SimpleTranslationEntries } from "#app/interfaces/locales";
+
+export const modifier: SimpleTranslationEntries = {
+  "surviveDamageApply": "{{pokemonNameWithAffix}} aguentou o tranco\nusando sua {{typeName}}!",
+  "turnHealApply": "{{pokemonNameWithAffix}} restaurou um pouco de PS usando\nseu {{typeName}}!",
```
Comment (josericardo-fo) [inline src/locales/pt_BR/modifier.ts:5]:
```suggestion
  "turnHealApply": "{{pokemonNameWithAffix}} restaurou um pouco de PS usando\nsuas {{typeName}}!",
```

File: src/locales/pt_BR/modifier.ts [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+  "hitHealApply": "{{pokemonNameWithAffix}} restaurou um pouco de PS usando\nsua {{typeName}}!",
+  "pokemonInstantReviveApply": "{{pokemonNameWithAffix}} foi revivido\npor seu {{typeName}}!",
+  "moneyInterestApply": "Você recebeu um juros de ₽{{moneyAmount}}\nde sua {{typeName}}!",
+  "turnHeldItemTransferApply": "{{itemName}} de {{pokemonNameWithAffix}} foi absorvido(a)\npelo {{typeName}} de {{pokemonName}}!",
+  "contactHeldItemTransferApply": "{{itemName}} de {{pokemonNameWithAffix}} foi pego(a)\npela {{typeName}} de {{pokemonName}}!",
+  "enemyTurnHealApply": "{{pokemonNameWithAffix}}\nrestaurou um pouco de PS!",
```
Comment (josericardo-fo) [inline src/locales/pt_BR/modifier.ts:11]:
```suggestion
  "enemyTurnHealApply": "{{pokemonNameWithAffix}}\nrestaurou um pouco de seus PS!",
```

File: src/locales/fr/modifier.ts [hunk 9]
```diff
@@ -0,0 +1,12 @@
...
+
+export const modifier: SimpleTranslationEntries = {
+  "surviveDamageApply": "{{pokemonNameWithAffix}} tient bon grâce à {{typeName}} !",
+  "turnHealApply": "Les PV du {{pokemonNameWithAffix}}\nsont un peu restaurés par {{typeName}} !",
+  "hitHealApply": "Les PV du {{pokemonNameWithAffix}}\nsont un peu restaurés par {{typeName}} !",
+  "pokemonInstantReviveApply": "{{pokemonNameWithAffix}} a repris connaissance\navec {{typeName}} et est prêt à se battre de nouveau !",
+  "moneyInterestApply": "Vous recevez {{moneyAmount}} ₽\nd’intérets de la {{typeName}} !",
+  "turnHeldItemTransferApply": "{{itemName}} de {{pokemonNameWithAffix}} est absorbé\npar le {{typeName}} de {{pokemonName}} !",
+  "contactHeldItemTransferApply": "{{itemName}} de {{pokemonNameWithAffix}} est volé\npar {{typeName}} de {{pokemonName}} !",
+  "enemyTurnHealApply": "{{pokemonNameWithAffix}}\nrestaure un peu ses PV !",
```
Comment (Adri1) [inline src/locales/fr/modifier.ts:11]:
```suggestion
  "surviveDamageApply": "{{pokemonNameWithAffix}} tient bon\ngrâce à son {{typeName}} !",
  "turnHealApply": "Les PV de {{pokemonNameWithAffix}}\nsont un peu restaurés par les {{typeName}} !",
  "hitHealApply": "Les PV de {{pokemonNameWithAffix}}\nsont un peu restaurés par le {{typeName}} !",
  "pokemonInstantReviveApply": "{{pokemonNameWithAffix}} a repris connaissance\navec sa {{typeName}} et est prêt à se battre de nouveau !",
  "moneyInterestApply": "La {{typeName}}vous rapporte\n{{moneyAmount}} ₽ d’intérêts !",
  "turnHeldItemTransferApply": "{{itemName}} de {{pokemonNameWithAffix}} est absorbé\npar le {{typeName}} de {{pokemonName}} !",
  "contactHeldItemTransferApply": "{{itemName}} de {{pokemonNameWithAffix}} est volé\npar {{typeName}} de {{pokemonName}} !",
  "enemyTurnHealApply": "{{pokemonNameWithAffix}}\nrestaure un peu ses PV !",
```

**Review by Adri1 — COMMENTED** [review]
Edited French

File: src/locales/fr/modifier.ts [hunk 10]
```diff
@@ -0,0 +1,12 @@
+import { SimpleTranslationEntries } from "#app/interfaces/locales";
+
+export const modifier: SimpleTranslationEntries = {
+  "surviveDamageApply": "{{pokemonNameWithAffix}} tient bon\ngrâce à son {{typeName}} !",
+  "turnHealApply": "Les PV de {{pokemonNameWithAffix}}\nsont un peu restaurés par les {{typeName}} !",
+  "hitHealApply": "Les PV de {{pokemonNameWithAffix}}\nsont un peu restaurés par le {{typeName}} !",
+  "pokemonInstantReviveApply": "{{pokemonNameWithAffix}} a repris connaissance\navec sa {{typeName}} et est prêt à se battre de nouveau !",
+  "moneyInterestApply": "La {{typeName}} vous rapporte\n{{moneyAmount}} ₽ d’intérêts !",
+  "turnHeldItemTransferApply": "{{itemName}} de {{pokemonNameWithAffix}} est absorbé\npar le {{typeName}} de {{pokemonName}} !",
+  "contactHeldItemTransferApply": "{{itemName}} de {{pokemonNameWithAffix}} est volé\npar l’{{typeName}} de {{pokemonName}} !",
```
Comment (Adri1) [inline src/locales/fr/modifier.ts:10]:
```suggestion
  "turnHeldItemTransferApply": "{{itemName}} de {{pokemonNameWithAffix}} est absorbé·e\npar le {{typeName}} de {{pokemonName}} !",
  "contactHeldItemTransferApply": "{{itemName}} de {{pokemonNameWithAffix}} est volé·e\npar l’{{typeName}} de {{pokemonName}} !",
```

**Review by Adri1 — COMMENTED** [review]
Final French Edit (in theory)

**Review by Adri1 — APPROVED** [review]

**Review by josericardo-fo — APPROVED** [review]
LGTM

**Review by 40chyan — APPROVED** [review]

**Review by mercurius-00 — APPROVED** [review]

**Review by Tempo-anon — APPROVED** [review]
Approved by the wiki/tl team

**Pull request merged**